metricbeat/module/envoyproxy/server/data.go

```diff
@@ -0,0 +1,187 @@
+				"downstream_cx_ssl_total":                       c.Int("downstream_cx_ssl_total"),
+				"downstream_cx_total":                           c.Int("downstream_cx_total"),
+				"downstream_cx_tx_bytes_buffered":               c.Int("downstream_cx_tx_bytes_buffered"),
+				"downstream_cx_tx_bytes_total":                  c.Int("downstream_cx_tx_bytes_total"),
+				"downstream_cx_websocket_active:":               c.Int("downstream_cx_websocket_active"),
+				"downstream_cx_websocket_total":                 c.Int("downstream_cx_websocket_total"),
+				"downstream_flow_control_paused_reading_total":  c.Int("downstream_flow_control_paused_reading_total"),
+				"downstream_flow_control_resumed_reading_total": c.Int("downstream_flow_control_resumed_reading_total"),
+				"downstream_rq_1xx":                             c.Int("downstream_rq_1xx"),
+				"downstream_rq_2xx":                             c.Int("downstream_rq_2xx"),
+				"downstream_rq_3xx":                             c.Int("downstream_rq_3xx"),
+				"downstream_rq_4xx":                             c.Int("downstream_rq_4xx"),
+				"downstream_rq_5xx":                             c.Int("downstream_rq_5xx"),
+				"downstream_rq_active":                          c.Int("downstream_rq_active"),
+				"downstream_rq_http1_total":                     c.Int("downstream_rq_http1_total"),
+				"downstream_rq_http2_total":                     c.Int("downstream_rq_http2_total"),
+				"downstream_rq_non_relative_path":               c.Int("downstream_rq_non_relative_path"),
+				"downstream_rq_response_before_rq_complete":     c.Int("downstream_rq_response_before_rq_complete"),
+				"downstream_rq_rx_reset":                        c.Int("downstream_rq_rx_reset"),
+				"downstream_rq_too_large":                       c.Int("downstream_rq_too_large"),
+				"downstream_rq_total":                           c.Int("downstream_rq_total"),
+				"downstream_rq_tx_reset":                        c.Int("downstream_rq_tx_reset"),
+				"downstream_rq_ws_on_non_ws_route":              c.Int("downstream_rq_ws_on_non_ws_route"),
+				"rs_too_large":                                  c.Int("rs_too_large"),
+			},
+			"async-client": s.Object{
+				"no_cluster":         c.Int("no_cluster"),
+				"no_route:":          c.Int("no_route"),
+				"rq_direct_response": c.Int("rq_direct_response"),
+				"rq_redirect":        c.Int("rq_redirect"),
+				"rq_total":           c.Int("rq_total"),
+			},
+		},
+	}
+)
+
+func eventMapping(response []byte) (common.MapStr, error) {
+	data := map[string]interface{}{}
+	var events common.MapStr
+	var err error
+
+	stats := []string{
+		"cluster_manager.*",
+		"filesystem.*",
+		"runtime.*",
+		"listener_manager.*",
+		"stats.*",
+		"server.*",
+		"listener.admin.*",
+		"listener.admin.http.admin.*",
+		"http.admin.*",
+		"http.async-client.*",
+	}
+	for i := 0; i < len(stats); i++ {
+		event := findStats(data, response, stats[i])
+		events, err = schema.Apply(event)
```

Comment:
Isn't this overriding the `events` var on every pass? It sounds like you could call it just once after the loop (using the loop to populate `data` only)

Reply:
Also, have a look to Apply options, the default behavior is: all fields are required, you will get an error if any of them is missing. I'm wondering if some fields may change between releases. A more relaxed check would be: `schema.Apply(event, schema.FailOnRequired)`. Pinging @jsoriano on this

Reply:
There is an example of required fields [here](https://github.com/elastic/beats/blob/14da4086cab07a522b4d5db2e5c9ed2ee906df73/libbeat/autodiscover/providers/jolokia/discovery.go#L59).

Reply:
Thanks for your review. I think now overriding issue is solved but I am not sure which fields are required. I am handle with this.